slugbuilder/builder/build.sh

```diff
@@ -57,6 +57,38 @@ curl() {
   $(which curl) --fail --silent --retry 3 $@
 }
 
+# removes leading and trailing whitespace
+trim() {
+  local var="$*"
+  var="${var#"${var%%[![:space:]]*}"}"
+  var="${var%"${var##*[![:space:]]}"}"
+  echo -n "$var"
+}
+
+prune_slugignore() {
+  shopt -s nullglob
+  # read slugignore into array
+  local globs=()
+  local paths=()
+  readarray -t globs < "${build_root}/.slugignore"
+  # for line in slugignore
+  for glob in ${globs[@]}; do
+    # strip whitespace
+    glob=$(trim ${glob})
+    # ignore blank lines and comment lines
+    if [[ ${glob} == "" ]] || [[ ${glob:0:1} == "#" ]]; then
+      continue
+    fi
+    # remove leading slash(es)
+    glob="${glob#"${glob%%[!"/"]*}"}"
+    # append to build root and add to array of paths to remove
+    paths=("${paths[@]}" ${build_root}/${glob})
+  done
+  echo_title "Deleting ${#paths[@]} files matching .slugignore patterns."
+  rm -f ${paths[@]}
+  shopt -u nullglob
+}
```

Comment:
Would be nice to add a BATS based test for this (see [release.bats](https://github.com/flynn/flynn/blob/master/script/test/release.bats) for an example), or perhaps right it in Go and call it with the `.slugignore` file as input :smile: